### PR TITLE
feat: Create service account and roles

### PR DESCRIPTION
**Pull request by SamoKopecky**
References #23, #24 
- `ServiceAccount` and `serviceAccountName` is already created in #19 
- Create a `role` and a `role-binding` for `peribolos-as-service` `ServiceAccount`.

### CODE REVIEW

File: manifests/role.yaml [hunk 1]
```diff
@@ -0,0 +1,21 @@
+kind: Role
+apiVersion: rbac.authorization.k8s.io/v1
+metadata:
+  name: peribolos-role
+  namespace: peribolos
+rules:
+- apiGroups:
+  - "tekton.dev"
+  resources:
+  - "taskRuns"
```
Comment (tumido) [inline manifests/role.yaml:10]: I think according to the specification this should reference the plural name of the resource

```suggestion
  - "taskruns"
```

image

File: manifests/role.yaml [hunk 2]
```diff
@@ -0,0 +1,21 @@
+kind: Role
+apiVersion: rbac.authorization.k8s.io/v1
+metadata:
+  name: peribolos-role
+  namespace: peribolos
```
Comment (tumido) [inline manifests/role.yaml:5]: Let's leave the namespace to be filled in by kustomize in here

File: manifests/role-binding.yaml [hunk 3]
```diff
@@ -0,0 +1,13 @@
+kind: RoleBinding
+apiVersion: rbac.authorization.k8s.io/v1
+metadata:
+  name: peribolos-role-binding
+  namespace: peribolos
```
Comment (tumido) [inline manifests/role-binding.yaml:5]: Let's leave the namespace to be filled in by kustomize in here

File: manifests/role-binding.yaml [hunk 4]
```diff
@@ -0,0 +1,13 @@
...
+subjects:
+- kind: ServiceAccount
+  name: peribolos-as-service
+  namespace: peribolos
```
Comment (tumido) [inline manifests/role-binding.yaml:9]: Kustomize can also change this value. Let's leave the namespace here to be filled in as well...

File: manifests/role.yaml (hunk 1 above)
Comment (SamoKopecky) [reply manifests/role.yaml:10]: Yes it should thanks!

**Comment by tumido** [comment]
/hold

Please don't recreate the PR, rather fix your history and force push. :slightly_smiling_face: 

I think a simple `git rebase main` should do the trick now and you should be left with 3 commits in this PR, then just `push --force` to this branch

**Review by tumido — COMMENTED** [review]
Code looks good, I like it. I'll approve once rebased :slightly_smiling_face:

**Comment by SamoKopecky** [comment]
Thanks for the tip! It should be fine now.

**Review by tumido — APPROVED** [review]
/lgtm
/approve

**Pull request merged**